### PR TITLE
:bug: Fixed import of produce from immer

### PR DESCRIPTION
**Pull request by ogregorio**
Hello friend,

I observed the import of the producer from the immer and it is following the old pattern, this was breaking the battery of tests, I fix the import.

What do you think?

Issue #178 

Screenshot from 2023-04-27 14-23-16

Screenshot from 2023-04-27 14-23-58

### CODE REVIEW

File: package.json
Comment (yezyilomo) [inline package.json:1]: This looks nice, but I wanted to check on immer version where the started supporting `import {produce} from immer` to make sure we don't skip any version. like is `9.0.12` the first version of immer to support it?.

File: src/State.ts
Comment (yezyilomo) [inline src/State.ts:1]: Looks nice!, a space before `produce` would be nice, just to be consistent.

**Review by yezyilomo — CHANGES_REQUESTED** [review]
Thank you for your PR, I've suggested few changes to make, I'd gladly merge it after making those changes.

**Comment by yezyilomo** [comment]
Ooooh and don't worry about the failing CI build, I think some things have been updated to GH actions, I'll work on it.

File: package.json
Comment (ogregorio) [reply package.json:1]: Desculpe, é um erro. A correção é v10.0.0.

**Review by yezyilomo — APPROVED** [review]
Thank you for your pull request and fast response. Your contribution is greatly appreciated.

**Pull request merged**